### PR TITLE
[SignalR TS] Improve connection error messages

### PR DESCRIPTION
**Pull request by BrennanConroy**
Start of https://github.com/dotnet/aspnetcore/issues/5350

Let the bike-shedding commence.

### CODE REVIEW

**Review by bradygaster — APPROVED** [review]

File: src/SignalR/clients/ts/signalr/src/ServerSentEventsTransport.ts [hunk 1]
```diff
@@ -90,11 +90,13 @@ export class ServerSentEventsTransport implements ITransport {
...
                 eventSource.onerror = (e: Event) => {
-                    const error = new Error("Error occurred while starting EventSource");
+                    // EventSource doesn't give any useful information about server side closes.
                     if (opened) {
-                        this._close(error);
+                        this._close();
                     } else {
-                        reject(error);
+                        reject(new Error("EventSource failed to connect. The connection could not be found on the server,"
+                        + " either the connection ID is not present on the server, or a proxy is refusing/buffering the connection."
```
Comment (davidfowl) [inline src/SignalR/clients/ts/signalr/src/ServerSentEventsTransport.ts:98]: 👏🏾  This error message is EF level good.

cc @ajcvickers

**Comment by davidfowl** [comment]
This is great!

**Review by davidfowl — APPROVED** [review]
Are we doing this for all clients?

**Comment by BrennanConroy** [comment]
> Are we doing this for all clients?

That's the general idea. Have to audit them all to see how good/bad they currently are, and what info we can get out of them.

Also why I didn't say "Fixes issue#"

**Pull request merged**